sim/src/main/scala/spinal/sim/VCSBackend.scala

```diff
@@ -0,0 +1,374 @@
+package spinal.sim
+
+import org.apache.commons.io.FileUtils
+import spinal.sim.vpi.SharedMemIface
+
+import java.io.{File, PrintWriter}
+import java.nio.file.{Files, Paths}
+import scala.sys.process.{Process, ProcessLogger}
+
+import scala.collection.mutable
+case class VCSFlags(
```

Comment:
Thanks for the contribution!

Idea: can splitting flags this way be useful for other simulators? It could look like this:

```scala
trait SimulatorFlags {
  val xxxFlags: mutable.ArrayBuffer[String] = mutable.ArrayBuffer[String]()

  final def withXxxFlags(flags: String*): this.type = ...
  final def withDefaultXxxFlags: this.type = this.withXxxFlags(defaultXxxFlags)...
  final def withDefaultFlags: this.type = this.withDefaultXxxFlags...

  protected def defaultXxxFlags: mutable.ArrayBuffer[String]
}
```

And each backend would define its own implementation:

```scala
case class VerilatorFlags extends SimulatorFlags {
  def defaultVlogFlags = ... // Flags for cc?
  def defaultVhdlFlags = ??? // cannot be used as Verilog cannot build Vhdl
  def defaultElabFlags = ... // Flags for ld?
  ...
```

Reply:
It's case by case, I think. This can be discussed further.

Reply:
To me, simulator flags are all different stuff for different environnement, i'm not sure there is much gain trying to make stuff common.
But my opinon isn't realy strong.

Reply:
Was just an idea, one™ should compare how simulators work.

Maybe not for this PR, I'll do that maybe later, maybe never ^^'